src/main/java/io/confluent/connect/jdbc/TableQuerier.java

```diff
@@ -107,4 +121,11 @@ public int compareTo(TableQuerier other) {
       return this.name.compareTo(other.name);
     }
   }
+
+  private boolean exceptionIndicatesStalePreparedStatement(Connection db, SQLException e) throws SQLException {
+    // On PostgreSQL, queries with wildcards will be invalidated on schema changes
+    // and throw this error message.
+    return e.getMessage().contains("cached plan must not change result type") &&
```

Comment:
I think the way you can make this work without requiring the runtime dependency for everyone is to add the dependency as `compile` or `provided` scope, check the class name at runtime and then only use the class in blocks where you've validated the class must be available at runtime. So roughly something like this:

```
catch (SQLException e) {
  if (e.getClass().getName() == "PSQLException") {
    PSQLException pge = (PSQLException) e;
    // check PSQLException-specific stuff
  }
}
```

Since you won't actually use the class unless the class name matches (which means it must be available in the classpath since it has been instantiated), this will work for instances that use PG and execute that block and other drivers which won't. This is basically a manual, runtime type check & validation that the class exists.

I'm not really sure if that's a good idea or not and how often this approach is used in practice. I think some users of libraries do similar things to detect which version of libraries they are running against to figure out what features they can use (or work around incompatibilities in the libraries). It just seems more reliable to me than relying on the message since messages seem more likely to change than the class name since the class name is used by "normal" users to catch exceptions.

I think I might have discussed this approach to handling cases like this with @gwenshap before. @gwenshap any input on this? Are there other examples you're aware of that use a similar solution?